substrate/client/chain-spec/src/lib.rs

```diff
@@ -16,19 +16,229 @@
 // You should have received a copy of the GNU General Public License
 // along with this program. If not, see <https://www.gnu.org/licenses/>.
 
-//! Substrate chain configurations.
-//!
-//! This crate contains structs and utilities to declare
-//! a runtime-specific configuration file (a.k.a chain spec).
+#![allow(rustdoc::private_intra_doc_links)]
+
+//! This crate includes structs and utilities for defining configuration files (known as chain
+//! specification) for both runtime and node.
+//!
+//! # Intro: Chain Specification
+//!
+//! The chain specification comprises parameters and settings that define the properties and an
+//! initial state of a chain. Users typically interact with the JSON representation of the chain
+//! spec. Internally, the chain spec is embodied by the [`GenericChainSpec`] struct, and specific
+//! properties can be accessed using the [`ChainSpec`] trait.
+//!
+//! In summary, although not restricted to, the primary role of the chain spec is to provide a list
+//! of well-known boot nodes for the blockchain network and the means for initializing the genesis
+//! storage. This initialization is necessary for creating a genesis block upon which subsequent
+//! blocks are built. When the node is launched for the first time, it reads the chain spec,
+//! initializes the genesis block, and establishes connections with the boot nodes.
+//!
+//! The JSON chain spec is divided into three main logical sections:
+//! - one section details general chain properties,
+//! - another explicitly or indirectly defines the genesis storage, which, in turn, determines the
+//!   genesis hash of the chain,
+//! - the third deals with the runtime code.
+//!
+//! The chain specification consists of the following fields:
+//!
+//! <table>
```

Comment:
Why don't construct the tables using the Markdown syntax?

```
//! | Chain spec key | Description |
//! |----------------|-------------|
//! | name           | The human readable name of the chain. |
//! | id             | The chain type of this chain (refer to [ChainType](enum.ChainType.html)) |
```

Reply:
I used it initially. But formatting was nightmare, so I decided to use HTML tags to have better control.

Reply:
I'm just curious. Why formatting was a nightmare?
Maybe it is because you was trying to keep the table right hand side aligned to the same column? If that is the reason, maybe you can just don't keep it aligned?

Reply:
IMO html here is a bit too verbose and unfriendly if you read the docs in raw text

Reply:
Just refer to this [commit](https://github.com/paritytech/polkadot-sdk/commit/6da1de6dfbfd81e830f0e8426ead0cbd516134ac).
 
`rustfmt` simply destroyed the table. HTML is just simpler.
If it is difficult to read in raw you can run `cargo doc --open`.

Reply:
I know I can run `cargo doc --open` 😃 but often devs reads stuff in the sources.

BTW was a suggestion